torch/distributions/constraints.py

```diff
@@ -91,17 +133,17 @@ class Constraint:
             when computing validity.
     """
 
-    is_discrete = False  # Default to continuous.
-    event_dim = 0  # Default to univariate.
+    is_discrete: ClassVar[bool] = False  # Default to continuous.
+    event_dim: ClassVar[int] = 0  # Default to univariate.
```

Comment:
For the abstract base class, wouldn't it make more sense to implement these as properties? Then we could get rid of a bunch of `type: ignore[override]` in the subclasses as well.

Reply:
Also, it would be neat to make the base class a `Protocol`.